### DISCUSSION TITLE
Pass viewdatadictionary in blockgrid and blocklist html extensions. Add index to block.

### DISCUSSION THREAD

**Discussion by SofusMunch** [umbraco/Umbraco-CMS#13385] [Features and ideas]
The current GetBlockListHtml and GetBlockGridHtml extensions don't allow passing viewdatadictionary to the partials. I use nested blocklists, for example in a slider blockgrid component:

![slider](https://user-images.githubusercontent.com/789813/201302384-533a10b5-6f98-46e0-b301-f38388839f36.JPG)

Since I need to pass some settings from the gridcomponent to the listcomponents I use viewdatadictionary. Unfortunately I can't use the GetBlockListHtml extension.

![slider_view](https://user-images.githubusercontent.com/789813/201302894-852cb558-343f-462f-8693-758f5f1d8c61.JPG)

Also since I need to set the first slide with an "active" class, I've changed the default blocklist view to add an index value:

![blockindex](https://user-images.githubusercontent.com/789813/201304300-834baf90-25fc-44bc-8afd-56c2cbc50204.JPG)

I haven't looked at the source code, but if possible I think it would be a good addition if a block had an index property.

**Comment by lassefredslund** [comment]
Hi @SofusMunch.

I think you are right, that this would be a nice improvement. I'll add it to up-for-grabs for now.
